ocaml/xapi/xapi_network_attach_helpers.ml

```diff
@@ -56,6 +57,13 @@ let assert_pif_disallow_unplug_not_set ~__context pif =
   if (Db.PIF.get_disallow_unplug ~__context ~self:pif) then
     raise (Api_errors.Server_error(Api_errors.pif_does_not_allow_unplug, [ Ref.string_of pif ]))
 
+let assert_pif_not_in_fcoe ~__context pif =
+  let device_name = Db.PIF.get_device ~__context ~self:pif in
+  let dbg = Context.string_of_task __context in
+  let capabilities = Net.Interface.get_capabilities dbg ~name:device_name in
```

Comment:
For the purpose of this change, it is at unplug stage so the pif data should already be updated with the capacities from the Net.Interface. So the check should be done based on the pif data (in xapi database), in stead of another call to the network interface.

Reply:
Seems make sense, I will check the code to find out how xapi synchronize the field value with the command line outputs.

Reply:
Though Db.PIF.capabilities is setted when creating or during a new scan, that means its value won't change if we do not trigger `PIF.scan`. After user change the interface capabilities, it's necessary for user to run `xe pif-scan` to update.